sql/src/main/java/io/crate/operation/reference/doc/lucene/RawCollectorExpression.java

```diff
@@ -41,6 +44,10 @@ public void startCollect(CollectorContext context) {
 
     @Override
     public BytesRef value() {
-        return visitor.source().toBytesRef();
+        try {
+            return CompressorFactory.uncompressIfNeeded(visitor.source()).toBytesRef();
```

Comment:
raw is currently in use by copy from / to, please check how/why it works there and if uncompressing it here would slow things down